libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1906,6 +1906,85 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_message(mavlink_command_long_t &pack
     return result;
 }
 
+
+bool GCS_MAVLINK::try_send_mission_message(enum ap_message id)
+{
+    AP_Mission *mission = get_mission();
+    if (mission == nullptr) {
+        return true;
+    }
+    switch (id) {
+    case MSG_CURRENT_WAYPOINT:
+        CHECK_PAYLOAD_SIZE(MISSION_CURRENT);
+        mavlink_msg_mission_current_send(chan, mission->get_current_nav_index());
+        return true;
+    case MSG_MISSION_ITEM_REACHED:
+        CHECK_PAYLOAD_SIZE(MISSION_ITEM_REACHED);
+        mavlink_msg_mission_item_reached_send(chan, mission_item_reached_index);
+        return true;
+    case MSG_NEXT_WAYPOINT:
+        CHECK_PAYLOAD_SIZE(MISSION_REQUEST);
+        queued_waypoint_send();
+        return true;
+    default:
+        return true;
+    }
+    return true;
+}
+
+void NOINLINE GCS_MAVLINK::send_hwstatus()
+{
+    mavlink_msg_hwstatus_send(
+        chan,
+        hal.analogin->board_voltage()*1000,
+        0);
+}
+
+bool GCS_MAVLINK::try_send_message(enum ap_message id)
+{
+    if (telemetry_delayed()) {
+        return false;
+    }
+    switch(id) {
+
+    case MSG_NEXT_PARAM:
+        CHECK_PAYLOAD_SIZE(PARAM_VALUE);
+        queued_param_send();
+        return true;
+
+    case MSG_HWSTATUS:
+        CHECK_PAYLOAD_SIZE(HWSTATUS);
+        send_hwstatus();
+        return true;
+
+    case MSG_CURRENT_WAYPOINT:
+    case MSG_MISSION_ITEM_REACHED:
+    case MSG_NEXT_WAYPOINT:
+        return try_send_mission_message(id);
+
+    case MSG_MAG_CAL_PROGRESS:
+        if (Compass *compass = get_compass()) {
+            compass->send_mag_cal_progress(chan);
+        }
+        return true;
+    case MSG_MAG_CAL_REPORT:
+        if (Compass *compass = get_compass()) {
+            compass->send_mag_cal_report(chan);
```

Comment:
I'll make a method like the others.

The bool-vs-void thing is interesting; does "true" indicate full-success or partial success?  This is generally why our "frontend-do-something-to-every-backend" methods generally return void, I think.  Nasty things happen in here if something continuously returns false.  I'm inclined to leave it void.